pkg/util/cluster/client.go

```diff
@@ -120,43 +109,48 @@ func PopulateAuthDetailsFromSecret(
 	secret *corev1.Secret,
 	useServiceAccount bool,
 ) error {
-	var exists bool
-
-	if useServiceAccount {
-		serviceAccountToken, exists := secret.Data[ServiceAccountTokenKey]
-		if !exists {
-			return fmt.Errorf("%q data is missing from secret", ServiceAccountTokenKey)
-		}
-		clusterConfig.BearerToken = string(serviceAccountToken)
-
-		if insecure {
-			clusterConfig.Insecure = true
-		} else {
-			clusterConfig.CAData, exists = secret.Data[CertificateAuthorityKey]
-			if !exists {
-				return fmt.Errorf("%q data is missing from secret and insecure is false", CertificateAuthorityKey)
-			}
-		}
+	if insecure {
+		clusterConfig.Insecure = true
 	} else {
-		clusterConfig.CertData, exists = secret.Data[ClientCertificateKey]
-		if !exists {
-			return fmt.Errorf("%q data is missing from secret", ClientCertificateKey)
+		// if federatedCluster.Spec.Insecure is false, the ca data is required
+		var caExists bool
+		if clusterConfig.CAData, caExists = getCADataFromSecret(secret); !caExists {
```

Comment:
Nit: the 3 new functions defined seem a little repetitive. 

Some alternative ways to write this line:
```suggestion
		if clusterConfig.CAData = secret.Data[common.ClusterCertificateAuthorityKey]; len(clusterConfig.CAData) == 0 {
```

or
```suggestion
		if data := secret.Data[common.ClusterCertificateAuthorityKey]; len(data) == 0 {
			return fmt.Errorf(...)
		} else {
			clusterConfig.CAData = data
		}
```

Reply:
The code has been updated.